federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -143,19 +143,20 @@ func StartControllers(s *options.CMServer, restClientCfg *restclient.Config) err
 		}
 		glog.Infof("Loading client config for service controller %q", servicecontroller.UserAgentName)
 		scClientset := federationclientset.NewForConfigOrDie(restclient.AddUserAgent(restClientCfg, servicecontroller.UserAgentName))
-		servicecontroller := servicecontroller.New(scClientset, dns, s.FederationName, s.ServiceDnsSuffix, s.ZoneName, s.ZoneID)
-		glog.Infof("Running service controller")
-		if err := servicecontroller.Run(s.ConcurrentServiceSyncs, wait.NeverStop); err != nil {
-			glog.Fatalf("Failed to start service controller: %v", err)
+		serviceController, err := servicecontroller.New(scClientset, dns, s.FederationName, s.ServiceDnsSuffix, s.ZoneName, s.ZoneID)
+		if err != nil {
+			glog.Fatalf("Failed to new service controller: %v", err)
```

Comment:
Replace "Failed to new service controller" by "Failed to instantiate service controller"